app/uk/gov/hmrc/tariffclassificationfrontend/connector/BindingTariffClassificationConnector.scala

```diff
@@ -75,7 +76,17 @@ class BindingTariffClassificationConnector @Inject()(appConfig: AppConfig, clien
   }
 
   def findEvents(reference: String, pagination: Pagination)(implicit hc: HeaderCarrier): Future[Paged[Event]] = {
-    val url = s"${appConfig.bindingTariffClassificationUrl}/cases/$reference/events?page=${pagination.page}&page_size=${pagination.pageSize}"
+    findFilteredEvents(reference,pagination, None)
+  }
+
+  def findFilteredEvents(reference: String, pagination: Pagination, onlyEventTypes: Option[Set[EventType]])(implicit hc: HeaderCarrier): Future[Paged[Event]] = {
+
+    val searchParam = onlyEventTypes match {
+      case Some(only) => s"case_reference=$reference${only.mkString("&type=","&type=","&")}"
```

Comment:
only means eventsType ?  :-)

Reply:
I guess case_reference=$reference can be extract it  to use it one 
1st -> returnValue = case_reference=$reference 
2nd -> returnValue & onlyEventTypes.map(o =>  s"&type=$o").mkString("")  

just an idea

Reply:
Simplified this based on comment, clearer and shorter now I think. Thanks